# HG changeset patch
# User StrikerTheHedgefox <mossj32@gmail.com>
# Date 1744872126 10800
#      Thu Apr 17 03:42:06 2025 -0300
# Node ID 128627054d8b04a42164c2d1c4810ce51048e4a4
# Parent  ebc6cb0488eeb311f2e0512e293fa45dc5dc0e6c
# EXP-Topic strikers_stuff
Fix infinite bouncing bug, hopefully.

diff -r ebc6cb0488ee -r 128627054d8b src/p_map.cpp
--- a/src/p_map.cpp	Thu Apr 17 02:08:57 2025 -0300
+++ b/src/p_map.cpp	Thu Apr 17 03:42:06 2025 -0300
@@ -3477,7 +3477,7 @@
 	}
 
 	// The amount of bounces is limited
-	if (mo->bouncecount>0 && --mo->bouncecount == 0)
+	if ((mo->GetDefault()->bouncecount > 0) && (--mo->bouncecount <= 0))
 	{
 		if (mo->flags & MF_MISSILE)
 			P_ExplodeMissile(mo, line, NULL);
@@ -3545,7 +3545,7 @@
 		|| ((BlockingMobj->player == NULL) && (!(BlockingMobj->flags3 & MF3_ISMONSTER)))
 		))
 	{
-		if (mo->bouncecount > 0 && --mo->bouncecount == 0) return false;
+		if ((mo->GetDefault()->bouncecount > 0) && (--mo->bouncecount <= 0)) return false;
 
 		if (mo->flags7 & MF7_HITTARGET)	mo->target = BlockingMobj;
 		if (mo->flags7 & MF7_HITMASTER)	mo->master = BlockingMobj;
diff -r ebc6cb0488ee -r 128627054d8b src/p_mobj.cpp
--- a/src/p_mobj.cpp	Thu Apr 17 02:08:57 2025 -0300
+++ b/src/p_mobj.cpp	Thu Apr 17 03:42:06 2025 -0300
@@ -1814,7 +1814,7 @@
 	}
 
 	// The amount of bounces is limited
-	if (bouncecount>0 && --bouncecount==0)
+	if ((GetDefault()->bouncecount > 0) && (--bouncecount <= 0))
 	{
 		if (flags & MF_MISSILE)
 			P_ExplodeMissile(this, NULL, NULL);
